src/restore.rs

```diff
@@ -131,10 +110,11 @@ pub fn restore(
                 // https://github.com/sourcefrog/conserve/issues/82
                 continue;
             }
-        } {
-            ui::show_error(&e);
+        };
+        monitor.restore_entry_result(&entry, &result);
```

Comment:
One thing I think about here is that if you use the DefaultMonitor from a test, then this will by default emit no error message? I guess you can see in the stats that some files were not restored. I guess this is OK although, possibly, the default should print some messages for problems? But then it would take some thought about how to avoid duplicating the nutmeg-based version.

Reply:
For Testing I would suggest implementing an own monitor validating the progress state callbacks.  
Imo using the stats which are returned should be fine right now.